smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -73,7 +73,12 @@ contract MixinPurchase is
       // refresh the cached value
       idTo = toKey.tokenId;
       _recordOwner(_recipient, idTo);
-      newTimeStamp = block.timestamp + expirationDuration;
+      // check for a non-expiring key
+      if (expirationDuration == type(uint).max) {
```

Comment:
Hum I thought you set `0` for the unlimited duration keys?

Reply:
nope, the non-expiring keys are using the same process, so they are set from `expirationDuration` which is `type(uint).max`. That way we continue to check validity the same way i.e. `block.timestamp < key.expirationTimestamp`